src/clients/java/src/client.zig

```diff
@@ -708,3 +742,165 @@ const JNIHelper = struct {
         return @ptrCast(address[0..@as(usize, @intCast(length))]);
     }
 };
+
+/// Helper for managing the `AttachCurrentThread`/`DetachCurrentThread` lifecycle
+/// when the JNI layer is unaware of when the native thread exits.
+/// https://developer.android.com/training/articles/perf-jni#threads
+const JNIThreadHelper = struct {
+    var tls_key: tls.Key = 0;
+    var mutex: std.Thread.Mutex = .{};
+
+    /// This function calls `AttachCurrentThreadAsDaemon` to attach the current native thread to
+    /// the JVM as a daemon thread. It also registers a callback to call `DetachCurrentThread`
+    /// when the thread exits (e.g., when the client is closed or evicted).
+    pub fn attach_current_thread_with_cleanup(jvm: *jni.JavaVM) *jni.JNIEnv {
+        // Create the thread-local storage key and the corresponding destructor
+        // function pointer **once** per JVM (may be concurrent).
+        if (@atomicLoad(tls.Key, &tls_key, .acquire) == 0) {
+            mutex.lock();
+            defer mutex.unlock();
+
+            if (tls_key == 0) {
+                // The destructor function will be called before the thread exits.
+                const key = tls.create_key(&destructor_callback);
```

Comment:
`assert(key != 0);`

Reply:
Thanks for pointing it out! I just discovered that `zero` is indeed a valid tls key, so we cannot use it as the unitialized flag!

I did a nice refactor, now using `std.once`, and added tests for the initialization case.